zebrad/src/components/sync.rs

```diff
@@ -116,6 +116,14 @@ pub(super) const BLOCK_DOWNLOAD_TIMEOUT: Duration = Duration::from_secs(15);
 /// failure loop.
 pub(super) const BLOCK_VERIFY_TIMEOUT: Duration = Duration::from_secs(180);
 
+/// Controls how long we wait for a transaction download request to complete.
+/// TODO: review value and rationale
+pub(super) const TRANSACTION_DOWNLOAD_TIMEOUT: Duration = Duration::from_secs(15);
+
+/// Controls how long we wait for a transaction verify request to complete.
+/// TODO: review value and rationale
+pub(super) const TRANSACTION_VERIFY_TIMEOUT: Duration = Duration::from_secs(180);
```

Comment:
We probably want to keep this interval reasonably long, so that transactions await their UTXOs.

If it's too short, then a transaction can fail due to missing UTXOs, then those UTXOs can arrive in the next block, but we won't re-verify the transaction, because we rejected it. (At least until we implement #2682.)

Reply:
We might also want to think about the mempool denial of service risk here - if the queue is 10 transactions long, and each transaction can wait for up to 180 seconds, and all the transactions are malicious, we can wait for up to 18 seconds per transaction.

Maybe 1 target block interval would be better here?
Maybe we need another way of re-accepting transactions that were rejected due to missing UTXOs?

I opened #2694 for this - it doesn't need to be dealt with in this PR.

Reply:
Thanks! I've moved the constants to `downloads.rs` (it doesn't make much sense for them to be here) and added some simple docs in a68bf1c4b041.